src/components/select/filterable-select/filterable-select.component.js

```diff
@@ -208,16 +220,17 @@ const FilterableSelect = React.forwardRef(
       );
 
       setSelectedValue((prevValue) => {
-        if (isControlled.current && prevValue && !newValue) {
-          setFilterText("");
-          setTextValue("");
+        if (isControlled.current && prevValue !== newValue) {
+          setMatchingText(newValue);
         }
 
         return newValue;
       });
 
       setHighlightedValue(newValue);
-    }, [value, defaultValue, onChange]);
+      // prevent value update on filter change
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why didn't you just include `setMatchingText` as a dependency here?

Reply:
I don't want it to be updated on every filterText change

Reply:
I didn't read the comment properly 😄